index.tsx

```diff
@@ -732,6 +737,36 @@ function anyScopeRegex(regex: RegExp) {
   }
 }
 
+function matchParagraph(source: string, state: MarkdownToJSX.State, prevCapturedString?: string) {
+  if (state._inline || state._simple) {
+    return null
+  }
+
+  if (prevCapturedString && !prevCapturedString.endsWith('\n')) { // don't match continuation of a line
+    return null
+  }
+
+  let match = ''
+
+  for (const line of source.split('\n')) {
+    // bail out on first sign of non-paragraph block
+    if (NON_PARAGRAPH_BLOCK_SYNTAXES.some(regex => regex.test(line))) {
+      break
+    }
+    match += line + '\n'
+    if (/^\s*$/.test(line)) {
```

Comment:
Would this be equivalent?
```suggestion
    if (!line.trim()) {
```

Reply:
Trim appears to be significantly faster, so let's go with that